File: Codeforces/A_82.cpp
// https://codeforces.com/contest/82/problem/A
// solved: 2013-08-28 
#include <iostream>
#include <vector>
using namespace std;
int main()
{

    long long int n, i, b;
    cin >> n;
    long long int sum = n - 5, c = 1;
    if (n > 5)
    {
        for (c = 2; sum > c * 5;)
        {
            sum -= c * 5;
            c *= 2;
        }
        b = (sum - 1) / c + 1;
    }
    else
        b = n;
    if (b == 1)
        cout << "Sheldon\n";
    else if (b == 2)
        cout << "Leonard\n";
    else if (b == 3)
        cout << "Penny\n";
    else if (b == 4)
        cout << "Rajesh\n";
    else if (b == 5)
        cout << "Howard\n";
    return 0;
}

File: Codeforces/A_82.java
/* package whatever; // don't place package name! */

import java.util.*;
import java.lang.*;
import java.io.*;

/* Name of the class has to be "Main" only if the class is public. */
public class A_82
{
	public static void main (String[] args) throws java.lang.Exception
	{
		Scanner scan = new Scanner(System.in);
		int n = scan.nextInt();
		int powerOfTwo = 1;
		String[] initQueue = {"Sheldon", "Leonard", "Penny", "Rajesh", "Howard"};
	//	int i = 0;
		if(n <= 5)
			n--;
		while(n > powerOfTwo * 5){
			n -= (powerOfTwo * 5);
			powerOfTwo *= 2;
		}
		n = n / (powerOfTwo);
		System.out.println(initQueue[n]);
		
	}
	
}